core/src/consensus/stake/actions.rs

```diff
@@ -26,13 +26,39 @@ use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
 use crate::client::ConsensusClient;
 use crate::consensus::{ConsensusMessage, ValidatorSet};
 
-const ACTION_TAG_TRANSFER_CCS: u8 = 1;
-const ACTION_TAG_DELEGATE_CCS: u8 = 2;
-const ACTION_TAG_REVOKE: u8 = 3;
-const ACTION_TAG_SELF_NOMINATE: u8 = 4;
-const ACTION_TAG_REPORT_DOUBLE_VOTE: u8 = 5;
-const ACTION_TAG_REDELEGATE: u8 = 6;
-const ACTION_TAG_CHANGE_PARAMS: u8 = 0xFF;
+
+#[repr(u8)]
+enum ActionTag {
+    TrasnferCCS = 1,
+    DelegateCCS = 2,
+    Revoke = 3,
+    SelfNominate = 4,
+    ReportDoubleVote = 5,
+    Redelegate = 6,
+    ChangeParams = 0xFF,
+}
+
+impl Encodable for ActionTag {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.append(self);
```

Comment:
You should convert `enum ActionTag` type to rlp-appendable value like `s.append(&(*self as u8))` with derive `Clone` and `Copy` trait for the `ActionTag`

Reply:
As @HoOngEe said, you should call `append` with casted value. This code is the reason for stack overflow since it calls rlp_append infinitely.